rocket-kafka/src/main.rs

```diff
@@ -7,28 +7,29 @@ extern crate kafka;
 use std::time::Duration;
 use std::error::Error;
 
+use rocket::State;
+
 use kafka::producer::{Producer, Record, RequiredAcks};
 use kafka::error::Error as KafkaError;
 
-fn send_message(message: &str) -> Result<(), KafkaError>
+fn send_message(
+    mut producer: State<Producer>,
+    message: &str
+) -> Result<(), KafkaError>
 {
-    let kafka_server = "rocket-kafka_kafka:9092";
     let topic = "topic";
-
-    let mut producer = try!(Producer::from_hosts(vec![kafka_server.to_owned()])
-        .with_ack_timeout(Duration::from_secs(5))
-        .with_required_acks(RequiredAcks::One)
-        .create());
-
     try!(producer.send(&Record::from_value(topic, message)));
 
     Ok(())
 }
 
 #[post("/message/<message>")]
-fn message(message: &str) -> &str {
+fn message<'a>(
```

Comment:
most of the time functions are named with a "verb-like" sentence to transcribe the action it does like `receive_message` , `treat_message` etc.